### PR TITLE
PR: Expand the LSB experiment #7 

### PR DESCRIPTION
**Pull request by tchojnacki**

### CODE REVIEW

File: code/projects/steg/src/lib/.cargo/config.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[build]
+rustflags = ["-C", "target-feature=+bulk-memory"]
```
Comment (tchojnacki) [inline code/projects/steg/src/lib/.cargo/config.toml:2]: Fixes the optimizer issue we faced last week (cc: @jakubzehner).

File: code/projects/steg/src/lib/src/lsb1.rs [hunk 2]
```diff
@@ -1,36 +1,49 @@
...
 
-    image.map_bytes(|byte, index| (byte & !1) | bits.get(index).copied().unwrap_or_default())
+    let mut data = Vec::with_capacity(image.data().len());
+    for (i, bytes) in image.data().chunks(RGBA_CHANNELS).enumerate() {
+        for (j, byte) in bytes[..3].iter().enumerate() {
+            data.push(byte & !1 | bits.get(i * 3 + j).copied().unwrap_or_default());
+        }
+        data.push(u8::MAX);
+    }
+    create_image_data(&data, image.width())
```
Comment (tchojnacki) [inline code/projects/steg/src/lib/src/lsb1.rs:37]: By @jakubzehner (taken from `map_bytes`).

File: code/projects/steg/src/lib/src/lsb1.rs [hunk 3]
```diff
@@ -1,36 +1,49 @@
...
-pub fn lsb1_embed_text(image: &AppImage, text: &str) -> AppImage {
+pub fn lsb1_embed_text(image: &ImageData, text: &str) -> ImageData {
     let mut bits = Vec::with_capacity(text.len() * ASCII_CHAR_BITS);
     for c in text.chars() {
         let mut value = c as u8 & ASCII_CHAR_MASK;
         for _ in 0..ASCII_CHAR_BITS {
-            bits.push(value & 1);
-            value >>= 1;
+            bits.push(value >> (ASCII_CHAR_BITS - 1));
+            value <<= 1;
```
Comment (tchojnacki) [inline code/projects/steg/src/lib/src/lsb1.rs:26]: Fixes invalid LSB embedding as found during today's meeting.

Comment (jakubzehner) [reply code/projects/steg/src/lib/src/lsb1.rs:26]: Good job!

**Pull request merged**